packages/svelte/src/internal/client/dom/blocks/svelte-element.js

```diff
@@ -42,98 +48,113 @@ export function element(anchor_node, tag_fn, is_svg, render_fn) {
 		// effect
 		e: null,
 		// parent
-		p: /** @type {import('#client').Block} */ (current_block),
-		// transition
-		r: null,
-		// type
-		t: DYNAMIC_ELEMENT_BLOCK
+		p: /** @type {import('#client').Block} */ (current_block)
 	};
 
 	hydrate_block_anchor(anchor_node);
-	let has_mounted = false;
 
-	/** @type {string} */
+	/** @type {string | null} */
 	let tag;
 
+	/** @type {string | null} */
+	let current_tag;
+
 	/** @type {null | Element} */
 	let element = null;
 
-	const element_effect = render_effect(
-		() => {
-			tag = tag_fn();
-			if (has_mounted) {
-				execute_effect(render_effect_signal);
-			}
-			has_mounted = true;
-		},
-		block,
-		false
-	);
-
-	// Managed effect
-	const render_effect_signal = render_effect(
-		() => {
-			// We try our best infering the namespace in case it's not possible to determine statically,
-			// but on the first render on the client (without hydration) the parent will be undefined,
-			// since the anchor is not attached to its parent / the dom yet.
-			const ns =
-				is_svg || tag === 'svg'
-					? namespace_svg
-					: is_svg === false || anchor_node.parentElement?.tagName === 'foreignObject'
-						? null
-						: anchor_node.parentElement?.namespaceURI ?? null;
-
-			const next_element = tag
-				? hydrating
-					? /** @type {Element} */ (current_hydration_fragment[0])
-					: ns
-						? document.createElementNS(ns, tag)
-						: document.createElement(tag)
-				: null;
-
-			const prev_element = element;
-			if (prev_element !== null) {
-				block.d = null;
-			}
+	/** @type {import('#client').Effect | null} */
+	let effect;
 
-			element = next_element;
-			if (element !== null && render_fn !== undefined) {
-				let anchor;
-				if (hydrating) {
-					// Use the existing ssr comment as the anchor so that the inner open and close
-					// methods can pick up the existing nodes correctly
-					anchor = /** @type {Comment} */ (element.firstChild);
-				} else {
-					anchor = empty();
-					element.appendChild(anchor);
-				}
-				render_fn(element, anchor);
-			}
+	let each_item_block = current_each_item_block;
 
-			const has_prev_element = prev_element !== null;
-			if (has_prev_element) {
-				remove(prev_element);
-			}
-			if (element !== null) {
-				insert(element, null, anchor_node);
-				if (has_prev_element) {
-					const parent_block = block.p;
-					swap_block_dom(parent_block, prev_element, element);
-				}
+	const wrapper = render_effect(() => {
+		const next_tag = tag_fn() || null;
+		if (next_tag === tag) return;
+
+		var previous_each_item_block = current_each_item_block;
+		set_current_each_item_block(each_item_block);
```

Comment:
needs explanation why it's necessary to set the current each item block, and why the initial each item block (set a few lines above) is important for this